components/eamxx/src/physics/iop_forcing/eamxx_iop_forcing_process_interface.cpp

```diff
@@ -0,0 +1,520 @@
+  }
+  if (iop_nudge_tq) {
+    qv_iop = m_iop_data_manager->get_iop_field("q").get_view<const Pack*>();
+    t_iop  = m_iop_data_manager->get_iop_field("T").get_view<const Pack*>();
+  }
+  if (iop_nudge_uv) {
+    u_iop = use_large_scale_wind ? m_iop_data_manager->get_iop_field("u_ls").get_view<const Pack*>()
+                                 : m_iop_data_manager->get_iop_field("u").get_view<const Pack*>();
+    v_iop  = use_large_scale_wind ? m_iop_data_manager->get_iop_field("v_ls").get_view<const Pack*>()
+                                  : m_iop_data_manager->get_iop_field("v").get_view<const Pack*>();
+  }
+
+  // Team policy and workspace manager for eamxx
+  const auto policy_iop = ESU::get_default_team_policy(m_num_cols, nlev_packs);
+
+  // Reset internal WSM variables.
+  m_workspace_mgr.reset_internals();
+
+  // Avoid implicit capture of this
+  auto wsm = m_workspace_mgr;
+  auto num_levs = m_num_levs;
+
+  // Apply IOP forcing
+  Kokkos::parallel_for("apply_iop_forcing", policy_iop, KOKKOS_LAMBDA (const MemberType& team) {
+    const int icol  =  team.league_rank();
+
+    auto ps_i = ps(icol);
+    auto u_i = Kokkos::subview(horiz_winds, icol, 0, Kokkos::ALL());
+    auto v_i = Kokkos::subview(horiz_winds, icol, 1, Kokkos::ALL());
+    auto T_mid_i = ekat::subview(T_mid, icol);
+    auto qv_i = ekat::subview(qv, icol);
+    auto Q_i = Kokkos::subview(Q, icol, Kokkos::ALL(), Kokkos::ALL());
+
+    auto ws = wsm.get_workspace(team);
+    uview_1d<Pack> ref_p_mid, ref_p_int, ref_p_del;
+    ws.take_many_contiguous_unsafe<3>({"ref_p_mid", "ref_p_int", "ref_p_del"},
+                                      {&ref_p_mid,  &ref_p_int,  &ref_p_del});
+
+    // Compute reference pressures and layer thickness.
+    // TODO: Allow geometry data to allocate packsize
+    auto s_ref_p_mid = ekat::scalarize(ref_p_mid);
+    auto s_ref_p_int = ekat::scalarize(ref_p_int);
+    Kokkos::parallel_for(Kokkos::TeamVectorRange(team, num_levs+1), [&](const int& k) {
+      s_ref_p_int(k) = hyai(k)*ps0 + hybi(k)*ps_i;
+      if (k < num_levs) {
+        s_ref_p_mid(k) = hyam(k)*ps0 + hybm(k)*ps_i;
+      }
+    });
+    team.team_barrier();
+    ColOps::compute_midpoint_delta(team, num_levs, ref_p_int, ref_p_del);
+    team.team_barrier();
+
+    if (iop_dosubsidence) {
+    // Compute subsidence due to large-scale forcing
+      advance_iop_subsidence(team, num_levs, dt, ps_i, ref_p_mid, ref_p_int, ref_p_del, omega, ws, u_i, v_i, T_mid_i, Q_i);
+    }
+
+    // Update T and qv according to large scale forcing as specified in IOP file.
+    advance_iop_forcing(team, num_levs, dt, divT, divq, T_mid_i, qv_i);
+
+    if (iop_coriolis) {
+      // Apply coriolis forcing to u and v winds
+      iop_apply_coriolis(team, num_levs, dt, target_lat, u_ls, v_ls, u_i, v_i);
+    }
+
+    // Release WS views
+    ws.release_many_contiguous<3>({&ref_p_mid, &ref_p_int, &ref_p_del});
+  });
+  Kokkos::fence();
```

Comment:
Let me check, not sure why I had initially included.

Reply:
I think I included previously because I was observing some values later on host, so I needed to ensure that all my computation from the previous kernel was complete. Now that everything is on device it should be ok to remove.

Reply:
Tested on GPU and fence is not needed. Removed.